internal/policies/mount/mount_test.go

```diff
@@ -0,0 +1,172 @@
+package mount_test
+
+import (
+	"context"
+	"os"
+	"os/user"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/adsys/internal/policies/mount"
+	"github.com/ubuntu/adsys/internal/testutils"
+)
+
+func TestNew(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		invalidPerm bool
+
+		wantErr bool
+	}{
+		"creates manager successfully": {},
+
+		"creation fails with invalid runDir permissions": {invalidPerm: true, wantErr: true},
+	}
+
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			runDir := t.TempDir()
+			if tc.invalidPerm {
+				err := os.Chmod(runDir, 0100)
+				require.NoError(t, err, "Expected to be able to change permissions but got an error instead.")
+			}
+
+			_, err := mount.New(mount.WithRunDir(runDir))
+			if tc.wantErr {
+				require.Error(t, err, "Expected an error when creating manager but got none.")
+				return
+			}
+			require.NoError(t, err, "Expected no error when creating manager but got one.")
+		})
+	}
+}
+
+func TestApplyPolicy(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		entries    string
+		objectName string
+		computer   bool
+
+		usrDirPerm os.FileMode
```

Comment:
See my comment on the unit tests. Only one case change the permission, so I would go for something which explains more the intention of "not able to write in userDir) (see also `usrDir` vs `userDir`)